astropy/constants/constant.py

```diff
@@ -31,7 +31,7 @@ class ConstantMeta(type):
     among other reasons).
     """
 
-    def __new__(mcls, name, bases, d):
+    def __new__(cls, name, bases, d):
```

Comment:
This is a metaclass - I think it still makes sense to keep that distinction, no? (Like for modeling)

Reply:
this was already discussed in https://github.com/astropy/astropy/pull/17314#discussion_r1827270940

Reply:
sounds like minimal refactoring of the one place where `mcls` is still used would get everyone to agree (albeit not everyone's first choice). Would that work for you ?

Reply:
> Shouldn't we at least be consistent and either use `mcls` or `cls` consistently for metaclasses?

If we adopt the usage of `mcls` in metaclasses as a project convention then we would have to manually find all metaclasses and manually update the parameter names. Furthermore, we must ensure maintainers stay vigilant during code review to prevent names violating the convention from entering our code. On one hand pull requests that modify metaclasses are rare, but on the other hand that just means that it is very easy to forget about this convention.

Alternatively, we can let Ruff automatically enforce using `cls` also in metaclasses, which is what most of our metaclasses are doing anyways.

Reply:
> which is what most of our metaclasses are doing anyways.

A quick git grep suggests, if anything, the reverse is the case, most metaclasses use `mcls`. That said, there clearly are metaclasses that use `cls`, and this PR makes all consistent in an easy way, so let's indeed just go with it.